tests/core/data/test_json_data_node.py

```diff
@@ -0,0 +1,103 @@
+from taipy.config.exceptions.exceptions import InvalidConfigurationId
+
+
+class TestJSONDataNode:
+    def test_create(self):
+        path = "data/node/path"
+        dn = JSONDataNode("foo_bar", Scope.PIPELINE, name="super name", properties={"default_path": path})
+        assert isinstance(dn, JSONDataNode)
+        assert dn.storage_type() == "json"
+        assert dn.config_id == "foo_bar"
+        assert dn.name == "super name"
+        assert dn.scope == Scope.PIPELINE
+        assert dn.id is not None
+        assert dn.parent_id is None
+        assert dn.last_edition_date is None
+        assert dn.job_ids == []
+        assert not dn.is_ready_for_reading
+        assert dn.path == path
+
+        with pytest.raises(InvalidConfigurationId):
+            dn = JSONDataNode(
+                "foo bar", Scope.PIPELINE, name="super name", properties={"default_path": path, "has_header": False}
+            )
+
+    def test_new_csv_data_node_with_existing_file_is_ready_for_reading(self):
+        not_ready_dn_cfg = Config.configure_data_node(
+            "not_ready_data_node_config_id", "json", default_path="NOT_EXISTING.json"
+        )
+        not_ready_dn = _DataManager._bulk_get_or_create([not_ready_dn_cfg])[not_ready_dn_cfg]
+        assert not not_ready_dn.is_ready_for_reading
+
+        path = os.path.join(pathlib.Path(__file__).parent.resolve(), "data_sample/example.json")
+        ready_dn_cfg = Config.configure_data_node("ready_data_node_config_id", "json", default_path=path)
+        ready_dn = _DataManager._bulk_get_or_create([ready_dn_cfg])[ready_dn_cfg]
+        assert ready_dn.is_ready_for_reading
+
+    def test_create_with_missing_parameters(self):
+        with pytest.raises(MissingRequiredProperty):
+            JSONDataNode("foo", Scope.PIPELINE, DataNodeId("dn_id"))
+        with pytest.raises(MissingRequiredProperty):
+            JSONDataNode("foo", Scope.PIPELINE, DataNodeId("dn_id"), properties={})
+
+    def test_read(self):
+        not_existing_csv = JSONDataNode("foo", Scope.PIPELINE, properties={"default_path": "WRONG.json"})
+        with pytest.raises(NoData):
+            assert not_existing_csv.read() is None
+            not_existing_csv.read_or_raise()
+        path = os.path.join(pathlib.Path(__file__).parent.resolve(), "data_sample/example.json")
+        data_node = JSONDataNode("bar", Scope.PIPELINE, properties={"default_path": path})
+        data = data_node.read()
+        assert isinstance(data, list)
+        assert len(data) == 3
+
+    def test_read_invalid_json(self):
+        path = os.path.join(pathlib.Path(__file__).parent.resolve(), "data_sample/invalid.json.txt")
+        dn = JSONDataNode("foo", Scope.PIPELINE, properties={"default_path": path})
+        with pytest.raises(ValueError):
+            dn.read()
+
+    def test_write(self, json_file):
+        json_dn = JSONDataNode("foo", Scope.PIPELINE, properties={"default_path": json_file})
+        data = {"a": 1, "b": 2, "c": 3}
```

Comment:
can we expand the test to value being string and boolean and other primitive types like float for example?